drivers/scale_by_host_template_id.go

```diff
@@ -0,0 +1,404 @@
+	}
+
+	if config.Action != "up" && config.Action != "down" {
+		return http.StatusBadRequest, fmt.Errorf("Invalid action %v", config.Action)
+	}
+
+	if config.Amount <= 0 {
+		return http.StatusBadRequest, fmt.Errorf("Invalid amount: %v", config.Amount)
+	}
+
+	if config.HostTemplateID == "" {
+		return http.StatusBadRequest, fmt.Errorf("HostTemplateID not provided")
+	}
+
+	if config.Min <= 0 {
+		return http.StatusBadRequest, fmt.Errorf("Minimum scale not provided/invalid")
+	}
+
+	if config.Max <= 0 {
+		return http.StatusBadRequest, fmt.Errorf("Maximum scale not provided/invalid")
+	}
+
+	if config.Min >= config.Max {
+		return http.StatusBadRequest, fmt.Errorf("Max must be greater than min")
+	}
+
+	if config.Action == "up" {
+		if config.DeleteOption != "" {
+			return http.StatusBadRequest, fmt.Errorf("Delete option not to be provided while scaling up")
+		}
+	}
+
+	if config.Action == "down" {
+		if config.DeleteOption != "mostRecent" && config.DeleteOption != "leastRecent" {
+			return http.StatusBadRequest, fmt.Errorf("Invalid delete option/Delete option missing %v", config.DeleteOption)
+		}
+	}
+
+	return http.StatusOK, nil
+}
+
+func (s *ScaleByHostTemplateIDDriver) Execute(conf interface{}, apiClient *client.RancherClient, reqBody interface{}) (int, error) {
+	var currNameSuffix, baseHostName, currCloneName, suffix string
+	var count, index, newHostScale, baseHostIndex int64
+	httpClient := &http.Client{
+		Timeout: time.Second * 10,
+	}
+
+	config := &model.ScaleByHostTemplateID{}
+	err := mapstructure.Decode(conf, config)
+	if err != nil {
+		return http.StatusInternalServerError, errors.Wrap(err, "Couldn't unmarshal config")
+	}
+
+	action := config.Action
+	amount := config.Amount
+	min := config.Min
+	max := config.Max
+	deleteOption := config.DeleteOption
+
+	cattleConfig := rConfig.GetConfig()
+	cattleURL := cattleConfig.CattleURL
+	u, err := url.Parse(cattleURL)
+	if err != nil {
+		panic(err)
+	}
+	cattleURL = strings.Split(cattleURL, u.Path)[0] + "/v2-beta"
+
+	filters := make(map[string]interface{})
+	filters["sort"] = "created"
+	filters["order"] = "desc"
+	hostCollection, err := apiClient.Host.List(&client.ListOpts{
+		Filters: filters,
+	})
+	if len(hostCollection.Data) == 0 {
+		return http.StatusBadRequest, fmt.Errorf("No hosts for scaling found")
+	}
+
+	hostScalingGroup := []client.Host{}
+	baseHostIndex = -1
+	for _, host := range hostCollection.Data {
+		if config.HostTemplateID != "" {
+			// add hosts with specified hostTemplateId
+			if host.HostTemplateId == config.HostTemplateID {
+				hostScalingGroup = append(hostScalingGroup, host)
+				if host.Driver != "" {
+					baseHostIndex = int64(len(hostScalingGroup)) - 1
+				}
+			}
+		}
+	}
+
+	if baseHostIndex == -1 && action == "up" {
+		return http.StatusBadRequest, fmt.Errorf("Cannot use custom hosts for scaling up")
+	}
+
+	count = 0
+
+	if action == "up" {
+		// Consider the least recently created as base host for cloning
```

Comment:
@cjellick since we are using the "hostTemplateId" not "hostTemplate", I think it still fall into the "clone" concept?